models/demos/yolov4/tests/yolov4_perfomant_webdemo.py

```diff
@@ -0,0 +1,385 @@
+        )
+        ttnn.end_trace_capture(device, self.tid, cq_id=0)
+        assert trace_input_addr == ttnn.buffer_address(self.input_tensor)
+
+        self.device = device
+
+        # More optimized run with caching
+        # if use_signpost:
+        #    signpost(header="start")
+
+    def get_region_boxes(self, boxes_and_confs):
+        print("Getting boxes from boxes and confs ...")
+        boxes_list = []
+        confs_list = []
+
+        for item in boxes_and_confs:
+            boxes_list.append(item[0])
+            confs_list.append(item[1])
+
+        # boxes: [batch, num1 + num2 + num3, 1, 4]
+        # confs: [batch, num1 + num2 + num3, num_classes]
+        boxes = torch.cat(boxes_list, dim=1)
+        confs = torch.cat(confs_list, dim=1)
+
+        return [boxes, confs]
+
+    def execute_yolov4_trace_2cqs_inference(self, tt_inputs_host=None):
+        tt_inputs_host = self.tt_inputs_host if tt_inputs_host is None else tt_inputs_host
+        ttnn.wait_for_event(1, self.op_event)
+        ttnn.copy_host_to_device_tensor(tt_inputs_host, self.tt_image_res, 1)
+        ttnn.record_event(1, self.write_event)
+        ttnn.wait_for_event(0, self.write_event)
+        # TODO: Add in place support to ttnn to_memory_config
+        self.input_tensor = ttnn.reshard(self.tt_image_res, self.input_mem_config, self.input_tensor)
+        ttnn.record_event(0, self.op_event)
+        ttnn.execute_trace(self.device, self.tid, cq_id=0, blocking=False)
+        ttnn.synchronize_devices(self.device)
+        output = self.test_infra.output_tensor
+
+        output_tensor1 = ttnn.to_torch(output[0])
+        output_tensor1 = output_tensor1.reshape(1, 40, 40, 255)
+        output_tensor1 = torch.permute(output_tensor1, (0, 3, 1, 2))
+
+        output_tensor2 = ttnn.to_torch(output[1])
+        output_tensor2 = output_tensor2.reshape(1, 20, 20, 255)
+        output_tensor2 = torch.permute(output_tensor2, (0, 3, 1, 2))
+
+        output_tensor3 = ttnn.to_torch(output[2])
+        output_tensor3 = output_tensor3.reshape(1, 10, 10, 255)
+        output_tensor3 = torch.permute(output_tensor3, (0, 3, 1, 2))
+
+        n_classes = 80
+
+        yolo1 = YoloLayer(
+            anchor_mask=[0, 1, 2],
+            num_classes=n_classes,
+            anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+            num_anchors=9,
+            stride=8,
+        )
+
+        yolo2 = YoloLayer(
+            anchor_mask=[3, 4, 5],
+            num_classes=n_classes,
+            anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+            num_anchors=9,
+            stride=16,
+        )
+
+        yolo3 = YoloLayer(
+            anchor_mask=[6, 7, 8],
+            num_classes=n_classes,
+            anchors=[12, 16, 19, 36, 40, 28, 36, 75, 76, 55, 72, 146, 142, 110, 192, 243, 459, 401],
+            num_anchors=9,
+            stride=32,
+        )
```

Comment:
This can be placed in the initialize fn right? Shouldn't be recreated every execution

Reply:
@tt-aho Thanks for the hint. We are working on optimizing this post-processing block, by moving it to run on device. Accordingly, the code will be refactored.